SourceX/dx.cpp

```diff
@@ -30,6 +30,72 @@ SDL_Surface *renderer_texture_surface = nullptr;
 /** 8-bit surface wrapper around #gpBuffer */
 SDL_Surface *pal_surface;
 
+#ifdef PIXEL_LIGHT
+struct POINT {
+	int x, y;
+	POINT(){}
+	POINT(int x1, int y1): x(x1), y(y1){}
+};
+
+SDL_Surface *tmp_surface;
+SDL_Surface *ui_surface;
+const int num_ellipses = 15;
+POINT eliSizes[num_ellipses];
+SDL_Texture *ellipsesTextures[num_ellipses];
+int lightReady = 0;
+
+void prepareLightColors()
+{
+	int orange = 0xffae00;
+	int darkorange = 0xd66b0d;
+	int blue = 0x0000ff;
+	int lightblue = 0x147ee0;
+	int darkblue = 0x000099;
+	int green = 0x00ff00;
+	int red = 0xff0000;
+	int white = 0xffffff;
+	int lime = 0xbfff00;
+	int lightorange = 0xff9500;
+	int lightyellow = 0xfff27a;
+
+	//others
+	lightColorMap["PLAYERLIGHT"] = white;

```

Comment:
This should just be a struct or a series of variables, then there won't be any look-ups and you won't need a prepareLightColors function.

```c++
// light_color.h
extern const int LIGHT_FIREARROW;
```

```c++
// light_colors.cpp
namespace {
constexpr int kBlue = 0x0000ff;
}

const int LIGHT_FIREARROW = kBlue;
```

Reply:
but I'd need an extern for each variable :P